swarm_dial.go

```diff
@@ -377,15 +378,16 @@ var ConnSetupTimeout = time.Minute * 5
 // dialConnSetup is the setup logic for a connection from the dial side. it
 // needs to add the Conn to the StreamSwarm, then run newConnSetup
 func dialConnSetup(ctx context.Context, s *Swarm, connC iconn.Conn) (*Conn, error) {
```

Comment:
@whyrusleeping: I'm not sure what the best solution for the deadline here. `SetDeadline` doesn't make sense for a stream-multiplexed connection (QUIC supports deadlines for streams as soon as https://github.com/lucas-clemente/quic-go/issues/514 is fixed, but I don't think this will help us here).

The call to `peerstream.AddConn` shouldn't take any time, so we only have to worry about the `newConnSetup` call. We're already passing the context there, so we could do try to implement there.
What do you think?